faunadb-scala/src/main/scala/faunadb/streaming/SnapshotEventFlowProcessor.scala

```diff
@@ -0,0 +1,80 @@
+package faunadb.streaming
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{Flow, SubmissionPublisher}
+
+import faunadb.values._
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+private [faunadb] class SnapshotEventFlowProcessor(loadDocument: () => Future[Value])(implicit ec: ExecutionContext)
+  extends SubmissionPublisher[Value] with Flow.Processor[Value, Value] {
+  private val log = LoggerFactory.getLogger(getClass)
+  private var subscription: Flow.Subscription = _
+  private var subscriber: Flow.Subscriber[_ >: Value] = _
+  private val initialized = new AtomicBoolean(false)
+  @volatile private var snapshotTS: String = _
+
+  // We do not request data from the publisher until we have one subscriber
+  override def subscribe(subscriber: Flow.Subscriber[_ >: Value]): Unit = {
+    if (this.subscriber == null) {
+      this.subscriber = subscriber
+      super.subscribe(subscriber)
+      requestOne()
+    } else
+      throw new IllegalStateException("SnapshotEventFlowProcessor can have only one subscriber")
+  }
+
+  override def onSubscribe(subscription: Flow.Subscription): Unit =
+    this.subscription = subscription
+
+  override def onNext(event: Value): Unit = {
+    if (initialized.get()) {
+      val eventTS = event("txn").get.toString
+      if (eventTS > snapshotTS) submit(event) // ignore event older than doc. snapshot
```

Comment:
Probably best comparing longs here?

Reply:
I am not sure, what do we gain by using `Long` here?
Is there an edge with comparing `String`?

Reply:
If there is, we didn't find it yet. We always compare txn time as long values.